lib/homepage.dart

```diff
@@ -47,11 +49,16 @@ class _MyHomePageState extends State<MyHomePage> {
     offers = [];
     webinars = [];
     super.initState();
-    getEvents();
-    getWebinars();
-    getOffers();
-    getRecentEventsPosters();
-    getAnnouncements();
+  }
```

Comment:
I moved all asynchronous functions to another function instead of `initState`, this is to prevent synchronous API calls (initState cannot be asynchronous)